File: Lunchtime_2018_04_P1.cpp
// code_report Solution
// 

#include <iostream>

using namespace std;

int main ()
{
   int n; cin >> n;

   while (n--) 
   {
      int x; cin >> x;
      cout << (x % 5 == 0 ? (x % 2 == 0 ? 0 : 1) : -1) << endl;
   }

   return 0;
}

File: Lunchtime_2018_04_P1.java
// code_report Solution
// 

import java.util.*;
import java.io.*;

public class Main {
    
     public static void main(String []args)
     {
        Scanner scan = new Scanner (System.in);
        int n = scan.nextInt ();
        
        while (n-- != 0)
        {
            int x = scan.nextInt ();
            System.out.println (x % 5 == 0 ? (x % 2 == 0 ? 0 : 1) : -1);
        }
     }
}
